app/src/main/java/com/example/tasky/feature_agenda/presentation/Agenda.kt

```diff
@@ -0,0 +1,115 @@
+package com.example.tasky.feature_agenda.presentation
+
+import android.content.Context
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeDrawingPadding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import com.example.tasky.AuthNavRoute
+import com.example.tasky.R
+import com.example.tasky.common.domain.error.DataError
+import com.example.tasky.common.presentation.CreateErrorAlertDialog
+import com.example.tasky.common.presentation.LoadingSpinner
+
+@Composable
+fun AgendaContent(
+    navController: NavController,
+    agendaViewModel: AgendaViewModel = hiltViewModel()
+) {
```

Comment:
I think I've commented on that before, but you're still coupling your nav controller and ViewModel to the screen composable which will not make it previewable. You can quickly check this by adding a `@Preview` composable, it won't work. 
Instead, better create a root composable which accepts these arguments that break the preview and pass down arguments that don't:
```
@Composable
fun AgendaRoot(
    navController: NavController,
    viewModel: AgendaViewModel = hiltViewModel()
) {
    // Initialize state, etc.
    
    AgendaContent(
        state = state,
        onAction = onAction
    )
}

@Composable
fun AgendaContent(
    state: AgendaState,
    onAction: (AgendaAction) -> Unit
) {
    // Actual UI
}
```
`AgendaAction` can be a sealed interface with all different user actions that could happen:
```
sealed interface AgendaAction {
    data object OnLogoutClick: AgendaAction
    data class OnCreateAgendaItemClick(val itemType: AgendaItemType): AgendaAction
    ...
}
```
After that, you can preview `AgendaContent` without issues, since it's not bound to a ViewModel or NavController anymore.

Reply:
I think I finally understand this concept. I will make improvements in next PR!